Favor marshal unmashal binare bare (#5799)
cosmos/cosmos-sdk: CHANGELOG.md

  to define their own concrete `MsgSubmitProposal` types.
  * The module now accepts a `Codec` interface which extends the `codec.Marshaler` interface by
  requiring a concrete codec to know how to serialize `Proposal` types.

> * (codec) Now we favor the use of MarshalBinaryBare instead of LengthPrefixed in all cases that is not needed.

Reviewer: ```suggestion
* (codec) [\#5799](https://github.com/cosmos/cosmos-sdk/pull/5799) Now we favor the use of MarshalBinaryBare instead of LengthPrefixed in all cases that is not needed.
```

> * (codec) [\#5799](https://github.com/cosmos/cosmos-sdk/pull/5799) Now we favor the use of MarshalBinaryBare instead of LengthPrefixed in all cases that is not needed.

Reviewer 2: ```suggestion
* (codec) [\#5799](https://github.com/cosmos/cosmos-sdk/pull/5799) Now we favor the use of `(Un)MarshalBinaryBare` instead of `(Un)MarshalBinaryLengthPrefixed` in all cases that are not needed.
```
